File: src/api/cabinets.ts
import client from "./client";

export const listCabinets = (params?: { meterId?: number }) =>
  client.get("/cabinets", { params }).then(r => r.data);

export const createCabinet = (data: any) =>
  client.post("/cabinets", data).then(r => r.data);

export const updateCabinet = (id: number, data: any) =>
  client.put(`/cabinets/${id}`, data).then(r => r.data);

export const deleteCabinet = (id: number) =>
  client.delete(`/cabinets/${id}`).then(r => r.data);

File: backend/app/api/cabinets.py
from fastapi import APIRouter, HTTPException
from ..repositories import cabinet_repo

router = APIRouter(prefix="/api/cabinets", tags=["柜子管理"])


@router.get("")
async def list_cabinets(meterId: int = None):
    return cabinet_repo.list_cabinets(meter_id=meterId)


@router.post("", status_code=201)
async def create_cabinet(data: dict):
    return cabinet_repo.create_cabinet(data)


@router.put("/{cabinet_id}")
async def update_cabinet(cabinet_id: int, data: dict):
    result = cabinet_repo.update_cabinet(cabinet_id, data)
    if not result:
        raise HTTPException(404, "柜子不存在")
    return result


@router.delete("/{cabinet_id}")
async def delete_cabinet(cabinet_id: int):
    cabinet_repo.delete_cabinet(cabinet_id)
    return {"ok": True}
